tests/http/models/test_genric_resource.py

```diff
@@ -0,0 +1,110 @@
+        resource = GenericResource({"$meta": meta_data})
+        assert resource._meta == Meta(**meta_data)
+
+    def test_generic_resource_box_functionality(self):
+        resource = GenericResource(id=1, name="test_resource", nested={"key": "value"})
+
+        assert resource.id == 1
+        assert resource.name == "test_resource"
+        assert resource.nested.key == "value"
+
+    def test_with_both_meta_and_response(self, meta_data):
+        response = Response(200, json={})
+        meta_data["response"] = response
+        meta_object = Meta(**meta_data)
+
+        resource = GenericResource(
+            data="test_data",
+            **{"$meta": meta_data},  # noqa: WPS445 WPS517
+        )
+
+        assert resource.data == "test_data"
+        assert resource._meta == meta_object
+
+    def test_dynamic_attribute_access(self):
+        resource = GenericResource()
+
+        resource.dynamic_field = "dynamic_value"
+        resource.nested_object = {"inner": "data"}
+
+        assert resource.dynamic_field == "dynamic_value"
+        assert resource.nested_object.inner == "data"
+
+
+class TestGenericResourceFromResponse:
+    @pytest.fixture
+    def meta_data_single(self):
+        return {"ignored": ["one"]}  # noqa: WPS226
+
+    @pytest.fixture
+    def meta_data_two_resources(self):
+        return {"pagination": {"limit": 10, "offset": 0, "total": 2}, "ignored": ["one"]}  # noqa: WPS226
+
+    @pytest.fixture
+    def meta_data_multiple(self):
+        return {"ignored": ["one", "two"]}  # noqa: WPS226
+
+    @pytest.fixture
+    def single_resource_data(self):
+        return {"id": 1, "name": "test"}
+
+    @pytest.fixture
+    def single_resource_response(self, single_resource_data, meta_data_single):
+        return Response(200, json={"data": single_resource_data, "$meta": meta_data_single})
+
+    @pytest.fixture
+    def multiple_resource_response(self, single_resource_data, meta_data_two_resources):
+        return Response(
+            200,
+            json={
+                "data": [single_resource_data, single_resource_data],
+                "$meta": meta_data_two_resources,
+            },
+        )
+
+    def test_malformed_meta_response(self):
+        with pytest.raises(TypeError, match=re.escape("Response $meta must be a dict.")):
+            _resource = GenericResource.from_response(Response(200, json={"data": {}, "$meta": 4}))
+
+    def test_single_resource(self, single_resource_response):
+        resource = GenericResource.from_response(single_resource_response)
+        assert resource.id == 1
+        assert resource.name == "test"
+        assert isinstance(resource._meta, Meta)
+        assert resource._meta.response == single_resource_response
+
+    def test_two_resources(self, multiple_resource_response, single_resource_data):
+        with pytest.raises(TypeError, match=r"Response data must be a dict."):
+            _resource = GenericResource.from_response(multiple_resource_response)
```

Comment:
```suggestion
           GenericResource.from_response(multiple_resource_response)
```

And, we can remove this ignore -> 

https://github.com/softwareone-platform/mpt-api-python-client/pull/5/files#diff-fa602a8a75dc9dcc92261bac5f533c2a85e34fcceaff63b3a3a81d9acde2fc52R34